.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,23 @@
+Thank you for contributing to JanusGraph.
+
+In order to streamline the review of the contribution we ask you
+to ensure the following steps have been taken:
+
+### For all changes:
+- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
+- [ ] Does your PR body contains #xyz where xyz is the issue number you are trying to resolve?
+- [ ] Has your PR been rebased against the latest commit within the target branch (typically ```master```)?
+- [ ] Is your initial contribution a single, squashed commit?
```

Comment:
Considering the difficulty we had merging a large CR in the past (we made sure to give @dylanht credit), it does make sense to allow multiple commits sometimes. Seems that it would make sense for hosted branches (major feature branches) more than PR by individuals.